### PR TITLE
test: run slow tests without race detector in CI

### PR DESCRIPTION
**Pull request by erikgrinaker**

### CODE REVIEW

**Review by marbar3778 — APPROVED** [review]

**Comment by erikgrinaker** [comment]
Well, that didn't really help much. Locally I'm seeing 70-90% reductions in test time, but then I also have an 8-core machine. GitHub actions uses 2 cores, I think.

Will try using default parallelism (2) in case the additional jobs just cause more overhead.

File: tree_random_test.go [hunk 1]
```diff
@@ -28,9 +30,12 @@ func TestRandomOperations(t *testing.T) {
 		957382170,
 	}
 
-	for _, seed := range seeds {
-		seed := seed
+	for i, seed := range seeds {
+		i, seed := i, seed
 		t.Run(fmt.Sprintf("Seed %v", seed), func(t *testing.T) {
+			if testing.Short() && i >= 2 {
```
Comment (marbar3778) [inline tree_random_test.go:36]: 😄

Comment (erikgrinaker) [reply tree_random_test.go:36]: Hm?

Comment (marbar3778) [reply tree_random_test.go:36]: Just a smile for going with shortening the amount of test cases.

**Review by marbar3778 — APPROVED** [review]
looks good to me. Thank you for taking the time to make this go faster.

**Comment by erikgrinaker** [comment]
Superseded by #338.

**Pull request merged**